tests/src/testMappedImage.cc

```diff
@@ -0,0 +1,73 @@
+// SPDX-FileCopyrightText: Deutsches Elektronen-Synchrotron DESY, MSK, ChimeraTK Project <chimeratk-support@desy.de>
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#define BOOST_TEST_MODULE MappedImageTest
+// Only after defining the name include the unit test header.
+#include <boost/test/included/unit_test.hpp>
+// boost unit_test needs to be included before serverBasedTestTools.h
+#include "MappedImage.h"
+
+#include <boost/test/unit_test.hpp>
+
+using namespace boost::unit_test_framework;
+using namespace ChimeraTK;
+
+BOOST_AUTO_TEST_SUITE(MappedImageTestSuite)
+
+BOOST_AUTO_TEST_CASE(testStructMapping) {
+  // this test shows an example how to map user-defined opaque structs onto a byte array
+  struct AStruct : public OpaqueStructHeader {
+    int a = 0;
+    float x = 0, y = 0;
+    AStruct() : OpaqueStructHeader(typeid(AStruct)) {}
+  };
+  unsigned len = 100;
+  std::vector<unsigned char> buf(len);
+  MappedStruct<AStruct> ms(buf);
+  auto* h = ms.header();
+  h->x = 4.;
+  BOOST_CHECK(h->totalLength == sizeof(AStruct));
+
+  MappedStruct<AStruct> ms1(buf, MappedStruct<AStruct>::InitData::No);
+  BOOST_CHECK(ms1.header()->x == 4.);
+}
+
+BOOST_AUTO_TEST_CASE(testMappedImage) {
+  // this test shows MappedImage usage
+  std::vector<uint8_t> buffer(100);
```

Comment:
The buffer is 100, but you only use 8 elements of 2 bytes. Does it have a header? Is the header offset being tested correctly?

Reply:
test now checks header also